src/ucs/stats/stats.c

```diff
@@ -460,6 +476,184 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+                     sizeof(*ucs_stats_context.aggrgt_sum_db));
+            ucs_stats_context.aggrgt_sum_db = ucs_realloc(
+                    ucs_stats_context.aggrgt_sum_db, total_alloc_size_bytes,
+                    (const char*)UCS_STATS_CLSID_TO_NAME_DB_NAME);
+        }
+
+        /* Initialize entry */
+        ucs_snprintf_safe(
+                ucs_stats_context
+                        .aggrgt_sum_db[ucs_stats_context.aggrgt_sum_db_size]
+                        .class_name,
+                sizeof(ucs_stats_context
+                               .aggrgt_sum_db[ucs_stats_context.aggrgt_sum_db_size]
+                               .class_name),
+                "%s", class_name);
+        ucs_stats_context.aggrgt_sum_db[ucs_stats_context.aggrgt_sum_db_size]
+                .aggregate_sum_class_offset =
+                ucs_stats_context.aggrgt_sum_db_last_offset;
+
+        /* Initialize return value */
+        offset = ucs_stats_context.aggrgt_sum_db_last_offset;
+
+        /* Return class_id */
+        *class_id = ucs_stats_context.aggrgt_sum_db_size;
+
+        /* Update global offset */
+        ucs_stats_context.aggrgt_sum_db_last_offset += num_counters_in_class;
+
+        /* Reallocate counters names db */
+        ucs_stats_context.aggrgt_cnt_names_db_size += num_counters_in_class;
+        total_alloc_size_bytes =
+                (ucs_stats_context.aggrgt_cnt_names_db_size *
+                 sizeof(*ucs_stats_context.aggrgt_cnt_names_db));
+        ucs_stats_context.aggrgt_cnt_names_db = ucs_realloc(
+                ucs_stats_context.aggrgt_cnt_names_db, total_alloc_size_bytes,
+                (const char*)UCS_STATS_AGGRGT_SUM_CNT_NAMES_DB);
+
+        cnt_agrgt_sum_index = offset;
+        ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+            ucs_list_for_each(temp_node, &filter_node->type_list_head,
+                              type_list) {
+                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index].name =
+                        node->cls->counter_names[counter_index];
+                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index]
+                        .class_name = node->cls->name;
+            }
+            cnt_agrgt_sum_index++;
+        }
+
+        ucs_stats_context.aggrgt_sum_db_size++;
+    }
+
+    pthread_mutex_unlock(&ucs_stats_context.lock);
+
+    return offset;
+}
+
+
+/**
+ * Returns the aggregate-sum of all counters from all interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [in]  node                      Pointer to the statistics node.
+ * @param [in]  sel                       Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters      Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum   The aggregate-sum of all counters of each class/type.
```

Comment:
the real name is `aggregate_sum`

Reply:
unhandled

Reply:
As far as I understand, this comment is handled.

Reply:
it's not handled: the name in doxygen is still not same as the real param name